src/pytorch_lightning/core/saving.py

```diff
@@ -171,13 +173,15 @@ def on_hpc_load(self, checkpoint: Dict[str, Any]) -> None:
 
 
 def _load_from_checkpoint(
-    cls: Union["pl.LightningModule", "pl.LightningDataModule"],
+    cls: Union[Type["ModelIO"], Type["pl.LightningModule"], Type["pl.LightningDataModule"]],
     checkpoint_path: Union[str, IO],
-    map_location: Optional[Union[Dict[str, str], str, torch.device, int, Callable]] = None,
+    map_location: Optional[
+        Union[str, Callable, torch.device, Dict[Union[str, torch.device], Union[str, torch.device]]]
+    ] = None,
     hparams_file: Optional[str] = None,
     strict: Optional[bool] = None,
     **kwargs: Any,
-) -> Any:
+) -> Union["ModelIO", "pl.LightningModule", "pl.LightningDataModule"]:
```

Comment:
this should never return ModelIO however, the implementation of `_load_from_checkpoint` in core.saving is such that this return type is forced to appease mypy (based on my understanding).